Source/WebCore/dom/Attr.cpp

```diff
@@ -101,9 +102,31 @@ ExceptionOr<void> Attr::setPrefix(const AtomString& prefix)
 
 ExceptionOr<void> Attr::setValue(const AtomString& value)
 {
-    if (RefPtr element = m_element.get())
-        return element->setAttribute(qualifiedName(), value, true);
-    else
+    if (RefPtr element = m_element.get()) {
+        auto verifiedValue = value;
+        if (document().requiresTrustedTypes()) {
+            AttributeTypeAndSink attributeTypeAndSink = trustedTypeForAttribute(element->nodeName(),
+            qualifiedName().localName(), element->namespaceURI(), qualifiedName().namespaceURI());
+            if (!attributeTypeAndSink.attributeType.isNull()) {
+                auto attributeValue = trustedTypesCompliantAttributeValue(*document().scriptExecutionContext(),
+                    attributeTypeAndSink.attributeType, value, attributeTypeAndSink.sink);
```

Comment:
I don’t think `attributeValue` is a good way to distinguish the result of `trustedTypesCompliantAttributeValue` from passed in `value` argument, since that is also an attribute value. I suggest `compliantValue`. Also, a `Document` is a `ScriptExecutionContext` so there is no need to call `scriptExecutionContext()`, in fact to avoid this mistake in future I think we should consider hiding/deleting it from `Document` to avoid compiling unnecessary extra code.
```suggestion
                auto compliantValue = trustedTypesCompliantAttributeValue(document(), type.attributeType, value, type.sink);
```